tendermint/src/error.rs

```diff
@@ -1,206 +1,213 @@
-    NoVoteFound,
-
-    /// No Proposal found during conversion
-    #[error("no proposal found")]
-    NoProposalFound,
-
-    /// Invalid AppHash length found during conversion
-    #[error("invalid app hash Length")]
-    InvalidAppHashLength,
-
-    /// Invalid PartSetHeader
-    #[error("invalid part set header")]
-    InvalidPartSetHeader,
-
-    /// Missing Header in Block
-    #[error("missing header field")]
-    MissingHeader,
-
-    /// Missing Data in Block
-    #[error("missing data field")]
-    MissingData,
-
-    /// Missing Evidence in Block
-    #[error("missing evidence field")]
-    MissingEvidence,
-
-    /// Missing Timestamp in Block
-    #[error("missing timestamp field")]
-    MissingTimestamp,
-
-    /// Invalid Block
-    #[error("invalid block")]
-    InvalidBlock,
-
-    /// Invalid first Block
-    #[error("invalid first block")]
-    InvalidFirstBlock,
-
-    /// Missing Version field
-    #[error("missing version")]
-    MissingVersion,
-
-    /// Invalid Header
-    #[error("invalid header")]
-    InvalidHeader,
-
-    /// Invalid first Header
-    #[error("invalid first header")]
-    InvalidFirstHeader,
-
-    /// Invalid signature in CommitSig
-    #[error("invalid signature")]
-    InvalidSignature,
-
-    /// Invalid validator address in CommitSig
-    #[error("invalid validator address")]
-    InvalidValidatorAddress,
-
-    /// Invalid Signed Header
-    #[error("invalid signed header")]
-    InvalidSignedHeader,
-
-    /// Invalid Evidence
-    #[error("invalid evidence")]
-    InvalidEvidence,
-
-    /// Invalid BlockIdFlag
-    #[error("invalid block id flag")]
-    BlockIdFlag,
-
-    /// Negative voting power
-    #[error("negative power")]
-    NegativePower,
-
-    /// Mismatch between raw voting power and computed one in validator set
-    #[error("mismatch between raw voting power ({raw}) and computed one ({computed})")]
-    RawVotingPowerMismatch {
-        /// raw voting power
-        raw: vote::Power,
-        /// computed voting power
-        computed: vote::Power,
-    },
-
-    /// Missing Public Key
-    #[error("missing public key")]
-    MissingPublicKey,
-
-    /// Invalid validator parameters
-    #[error("invalid validator parameters")]
-    InvalidValidatorParams,
-
-    /// Invalid version parameters
-    #[error("invalid version parameters")]
-    InvalidVersionParams,
-
-    /// Negative max_age_num_blocks in Evidence parameters
-    #[error("negative max_age_num_blocks")]
-    NegativeMaxAgeNum,
-
-    /// Missing max_age_duration in evidence parameters
-    #[error("missing max_age_duration")]
-    MissingMaxAgeDuration,
-
-    /// Proposer not found in validator set
-    #[error("proposer with address '{}' not found in validator set", _0)]
-    ProposerNotFound(account::Id),
-}
+    InvalidHashSize
+    [DisplayError<Error>]
+    |_| { format_args!("invalid hash: expected hash size to be 32 bytes") },
+
+    NoTimestamp
+    [DisplayError<Error>]
```

Comment:
There is no need for any error source here, since there is no additional information needed.

Reply:
I have reset this change.